vm/src/vm.rs

```diff
@@ -823,6 +841,44 @@ impl VirtualMachine {
                 self.push_block(Block::TryExcept { handler: *handler });
                 None
             }
+            bytecode::Instruction::SetupWith { end } => {
+                let obj = self.pop_value();
+                // Call enter:
+                match self.call_method(obj, "__enter__", vec![]) {
+                    Ok(manager) => {
+                        self.push_block(Block::With {
+                            end: *end,
+                            context_manager: manager.clone(),
+                        });
+                        self.push_value(manager);
+                        None
+                    }
+                    Err(err) => Some(Err(err)),
+                }
+            }
+            bytecode::Instruction::CleanupWith { end: end1 } => {
+                let block = self.pop_block().unwrap();
+                if let Block::With {
+                    end: end2,
+                    context_manager,
+                } = &block
+                {
+                    assert!(end1 == end2);
+
+                    // call exit now:
+                    let exc_type = self.ctx.none();
```

Comment:
Worth adding a TODO regarding exception handling here?

Reply:
Yes, I will add a todo, and add exception handling in another PR.